### PR TITLE
update python and django to 3.6 and 2.0

### LINKED ISSUE #457: Migrate to Python 3.6 and Django 2.0
with all other depdencies to latest compatible version.

### PR DESCRIPTION
**Pull request by satya7289**
# Description
Successfully update the python and Django to 3.6 and 2.0 respectively.

Fixes #457

# Type of Change:

- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)

# How Has This Been Tested?
I have tested it locally, and it is working good to me.

# Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have made corresponding changes to the documentation

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by yashshah2820** [comment]
@satya7289 why are you adding a name to the apps this PR is only for migrating to Django 2.0 and Python 3.6+, please stick to that

File: vms/administrator/models.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ class Administrator(models.Model):
             RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)]+$', ),
         ],
     )
-    city = models.ForeignKey(City, null=True, blank=True)
-    state = models.ForeignKey(Region, null=True, blank=True)
-    country = models.ForeignKey(Country, null=True, blank=True)
+    city = models.ForeignKey(City, null=True, blank=True, on_delete=models.CASCADE)
+    state = models.ForeignKey(Region, null=True, blank=True, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/administrator/models.py:28]: please remove these changes

File: vms/administrator/models.py [hunk 2]
```diff
@@ -38,10 +38,10 @@ class Administrator(models.Model):
         ],
     )
     # Organization to Volunteer is a one-to-many relationship
-    organization = models.ForeignKey(Organization)
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
     # EmailField automatically checks if email address is a valid format
     email = models.EmailField(max_length=45, unique=True)
-    user = models.OneToOneField(User)
```
Comment (yashshah2820) [inline vms/administrator/models.py:44]: remove these changes

File: vms/event/models.py [hunk 3]
```diff
@@ -26,9 +26,9 @@ class Event(models.Model):
         blank=True,
         null=True,
     )
-    city = models.ForeignKey(City, null=True, blank=True)
-    state = models.ForeignKey(Region, null=True, blank=True)
-    country = models.ForeignKey(Country, null=True, blank=True)
+    city = models.ForeignKey(City, null=True, blank=True, on_delete=models.CASCADE)
+    state = models.ForeignKey(Region, null=True, blank=True, on_delete=models.CASCADE)
+    country = models.ForeignKey(Country, null=True, blank=True, on_delete=models.CASCADE)
     venue = models.CharField(
```
Comment (yashshah2820) [inline vms/event/models.py:32]: please remove these changes

File: vms/shift/models.py [hunk 4]
```diff
@@ -27,9 +27,9 @@ class Shift(models.Model):
         null=True,
     )
 
-    city = models.ForeignKey(City, null=True, blank=True)
-    state = models.ForeignKey(Region, null=True, blank=True)
-    country = models.ForeignKey(Country, null=True, blank=True)
+    city = models.ForeignKey(City, null=True, blank=True, on_delete=models.CASCADE)
+    state = models.ForeignKey(Region, null=True, blank=True, on_delete=models.CASCADE)
+    country = models.ForeignKey(Country, null=True, blank=True, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/shift/models.py:32]: please remove these changes

File: vms/shift/models.py [hunk 5]
```diff
@@ -39,7 +39,7 @@ class Shift(models.Model):
         null=True,
     )
     # Job to Shift is a one-to-many relationship
-    job = models.ForeignKey(Job)
+    job = models.ForeignKey(Job, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/shift/models.py:42]: remove these changes as well

File: vms/volunteer/models.py [hunk 6]
```diff
@@ -29,9 +29,9 @@ class Volunteer(models.Model):
             RegexValidator(r'^[(A-Z)|(a-z)|(0-9)|(\s)|(\-)|(\.)|(,)|(\:)]+$', ),
         ],
     )
-    city = models.ForeignKey(City, null=True, blank=True)
-    state = models.ForeignKey(Region, null=True, blank=True)
-    country = models.ForeignKey(Country, null=True, blank=True)
+    city = models.ForeignKey(City, null=True, blank=True, on_delete=models.CASCADE)
+    state = models.ForeignKey(Region, null=True, blank=True, on_delete=models.CASCADE)
+    country = models.ForeignKey(Country, null=True, blank=True, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/volunteer/models.py:34]: remove these changes

File: vms/volunteer/models.py [hunk 7]
```diff
@@ -43,7 +43,7 @@ class Volunteer(models.Model):
         ],
     )
     # Organization to Volunteer is a one-to-many relationship
-    organization = models.ForeignKey(Organization, null=True)
+    organization = models.ForeignKey(Organization, null=True, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/volunteer/models.py:46]: remove these changes

File: vms/volunteer/models.py [hunk 8]
```diff
@@ -78,7 +78,7 @@ class Volunteer(models.Model):
                     MinValueValidator(1)],
         blank=True)
 
-    user = models.OneToOneField(User)
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
```
Comment (yashshah2820) [inline vms/volunteer/models.py:81]: please remove these changes, these are not needed here

File: vms/administrator/models.py (hunk 1 above)
Comment (satya7289) [reply vms/administrator/models.py:28]: On upgrading django to 2.0 it is necessary to add on_delete method. For django 1.1 default is cascade therefore I add on_delete method to model cascade.
You can check out in django documentation also.

Comment (satya7289) [reply vms/administrator/models.py:28]: @yashshah2820

**Comment by satya7289** [comment]
> @satya7289 why are you adding a name to the apps this PR is only for migrating to Django 2.0 and Python 3.6+, please stick to that

It is required due to upgrading django to Django 2.0

**Comment by Harirai** [comment]
@yashshah2820 FYI, `on_delete` is mandatory in Django 2.0, and please refrain from flooding reviews with unnecessary comments.
Like a single comment, "I think changes with `on_delete` are not required" is sufficient.
Please ask the person "why are these changes made" instead of "remove them".

**Comment by yashshah2820** [comment]
@Harirai I will take a look at that for sure but I don't feel these comments are unnecessary as a PR review is done in this fashion though you point of having ``on_delete`` is correct.

**Comment by yashshah2820** [comment]
@satya7289 why are you adding ``app_name`` to the views, it is not needed

**Comment by satya7289** [comment]
> @satya7289 why are you adding `app_name` to the views, it is not needed

If I do not add this then namespace error will give, I don't remember the exact error message. I found the easiest way to handle these errors. 
You can check out the error message by locally fetching this PR, and remove the app_name from home app's url.

**Comment by yashshah2820** [comment]
I did run the PR on my machine and I commented all the ``app_name``, the server is running perfectly and I didn't encounter any error 🚀

**Comment by satya7289** [comment]
> I did run the PR on my machine and I commented all the `app_name`, the server is running perfectly and I didn't encounter any error

Screenshot from 2020-05-21 23-18-12

See this error I am talking about. 
Have you updated your virtualenv setup for Django2.0 and python3.6?

**Comment by yashshah2820** [comment]
Yes, I forgot that lucky me 😄, great work @satya7289

**Comment by yashshah2820** [comment]
@anitab-org/qa-team This PR is ready to be tested

**Comment by rpattath** [comment]
> > I did run the PR on my machine and I commented all the `app_name`, the server is running perfectly and I didn't encounter any error
> 
> Screenshot from 2020-05-21 23-18-12
> 
> See this error I am talking about.
> Have you updated your virtualenv setup for Django2.0 and python3.6?

I am hitting this error when I try to run the server locally. Does the [doc](https://github.com/satya7289/vms/blob/update-python-djnago/aut_docs/Installation_Setup.md) need any updates?

**Comment by rpattath** [comment]
I was able to get through the above issue using python 3.8 for venv. Please update this [doc](https://github.com/anitab-org/vms/blob/develop/aut_docs/Installation_Setup.md#a-one-time-setup) as well.

After the server ran successfully I attempted to sign-up as a non-admin user. When trying to click the email verification link I see the following. This was something that was working earlier, wondering if this issue is related to this PR.

Screenshot from 2020-05-22 15-54-23

**Comment by satya7289** [comment]
> I was able to get through the above issue using python 3.8 for venv. Please update this [doc](https://github.com/anitab-org/vms/blob/develop/aut_docs/Installation_Setup.md#a-one-time-setup) as well.
> 
> After the server ran successfully I attempted to sign-up as a non-admin user. When trying to click the email verification link I see the following. This was something that was working earlier, wondering if this issue is related to this PR.
> 
> Screenshot from 2020-05-22 15-54-23

For python3.6 it is working also, no need to update.

If you remove the binary string from the URL, then it will work. i.e. instead of this http://localhost:8000/registration/signup/activate/b'OA'/5gq-b1de6d2c0463dc27c11f/

use this http://localhost:8000/registration/signup/activate/OA/5gq-b1de6d2c0463dc27c11f/

**Comment by rpattath** [comment]
> > I was able to get through the above issue using python 3.8 for venv. Please update this [doc](https://github.com/anitab-org/vms/blob/develop/aut_docs/Installation_Setup.md#a-one-time-setup) as well.
> > After the server ran successfully I attempted to sign-up as a non-admin user. When trying to click the email verification link I see the following. This was something that was working earlier, wondering if this issue is related to this PR.
> > Screenshot from 2020-05-22 15-54-23
> 
> For python3.6 it is working also, no need to update.
> 
> If you remove the binary string from the URL, then it will work. i.e. instead of this http://localhost:8000/registration/signup/activate/b'OA'/5gq-b1de6d2c0463dc27c11f/
> 
> use this http://localhost:8000/registration/signup/activate/OA/5gq-b1de6d2c0463dc27c11f/

@satya7289 A little confused how a user/tester will  know how to remove the binary from the string. This is first time this has happened to me when I clicked the verification link. Please pardon my ignorance.

@yashshah2820 based on the current [workflow](https://anitab-org.zulipchat.com/#narrow/stream/216321-coding/topic/PR.20Labeling/near/198533824) the PR should be code reviewed and approved by at least 2 contributor before the label is changed to Needs Testing. So I am changing the PR back to Needs Review.

**Comment by satya7289** [comment]
1. URL doesn't parse binary or bytes. 
2. Here from b'OA' remove b'' like OA.
3. I guess this confirmation link is send to mail no need to copy paste from terminal. I checked in the developer mode that after you register you will able to login as volunteer with your credentials.
@rpattath

**Pull request merged**